crates/cxx-qt-gen/src/parser/cxxqtdata.rs

```diff
@@ -4,15 +4,23 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
 use crate::parser::{qobject::ParsedQObject, signals::ParsedSignalsEnum};
+use crate::syntax::attribute::attribute_tokens_to_value;
+use crate::syntax::foreignmod::{foreign_mod_to_foreign_item_types, verbatim_to_foreign_mod};
 use crate::syntax::{
     attribute::{attribute_find_path, attribute_tokens_to_ident},
     path::path_to_single_ident,
 };
 use std::collections::BTreeMap;
-use syn::{spanned::Spanned, Error, Ident, Item, ItemEnum, ItemImpl, Result, Type, TypePath};
+use syn::{
+    spanned::Spanned, Error, Ident, Item, ItemEnum, ItemImpl, LitStr, Result, Type, TypePath,
+};
 
 #[derive(Default)]
 pub struct ParsedCxxQtData {
+    /// Map of the cxx_name of any types defined in CXX extern blocks
+    ///
+    /// This is used in the C++ generation to map the Rust type name to the C++ name
+    pub cxx_names_map: BTreeMap<String, String>,
```

Comment:
Could have a namespace_map as well as a cxx_names_map. This could then take a type or method name and provide the namespace for it.

Reply:
Started splitting this into a separate pull request #390